core/templates/pages/exploration-player-page/modals/display-hint-modal.component.ts

```diff
@@ -33,12 +34,12 @@ import { PlayerTranscriptService } from '../services/player-transcript.service';
   templateUrl: './display-hint-modal.component.html'
 })
 export class DisplayHintModalComponent {
-  index: number;
-  COMPONENT_NAME_HINT: string;
-  hint: SubtitledHtml;
-  displayedCard: StateCard;
-  recordedVoiceovers: RecordedVoiceovers;
-  hintContentId: string;
+  index: number | undefined;
+  COMPONENT_NAME_HINT: string = AppConstants.COMPONENT_NAME_HINT;
+  hint!: SubtitledHtml | null;
+  displayedCard!: StateCard;
+  recordedVoiceovers!: RecordedVoiceovers;
+  hintContentId!: string | null;
```

Comment:
Why is `null` allowed here? Also why we need `!`?

Reply:
We need `!` here because it's undefined before calling `ngOnInit()` function. `null` is allowed here for `hint` because displayHint() funtion in [hints-and-solution-manager.service.ts file](https://github.com/oppia/oppia/blob/03bb986c0374f7412fbff94119b1081014b8884e/core/templates/pages/exploration-player-page/services/hints-and-solution-manager.service.ts) can return `null`, & for `hintContentId` because in `SubtitledHtml` class in the [subtitled-html.model.ts file](https://github.com/oppia/oppia/blob/03bb986c0374f7412fbff94119b1081014b8884e/core/templates/domain/exploration/subtitled-html.model.ts) the type of `contentID` is defined as `string | null`. Is there any change I can do? Thanks.

Reply:
Okay, please add explanatory comments into the code.

Reply:
Done!

Reply:
Hmm, thinking about the context here, we are in display hint modal, so we probably know the hint exists, should we instead create an error when the hint is null and thus here the hint will always be non-null?

Reply:
So, What should `displayHint()` return if `index >= this.numHintsReleased` in the `displayHint()` function in the [hints-and-solution-manager.service.ts file](https://github.com/oppia/oppia/blob/03bb986c0374f7412fbff94119b1081014b8884e/core/templates/pages/exploration-player-page/services/hints-and-solution-manager.service.ts)?

Reply:
Sorry, not sure what you mean? How is this related to my comment?

Reply:
Actually I wanted to ask how and where should I create the error? Should I create it in the [hints-and-solution-manager.service.ts file](https://github.com/oppia/oppia/blob/03bb986c0374f7412fbff94119b1081014b8884e/core/templates/pages/exploration-player-page/services/hints-and-solution-manager.service.ts) or display-hint-modal.component.ts file?

Reply:
Basically is it possible to somehow get rid of the `null` here? My understanding is: This is a component used for displaying hint, so the hint should always be set if we are in this component. 

I don't know where exactly put the errors, that need to be thought through.

Reply:
@vojtechjelinek Actually, I recently switched from Linux to Mac. I am getting https://github.com/oppia/oppia/discussions/15222#discussion-3965111 this error while setting up the environment. So, I am unable to run tests on my machine. What should I do?